influxql/engine.go

```diff
@@ -744,6 +746,54 @@ func ReduceMax(key Key, values []interface{}, e *Emitter) {
 	}
 }
 
+// MapStddev collects the values to pass to the reducer
+func MapStddev(itr Iterator, e *Emitter, tmax int64) {
+	var values []float64
+
+	for k, v := itr.Next(); k != 0; k, v = itr.Next() {
+		values = append(values, v.(float64))
+	}
+	if len(values) > 0 {
+		e.Emit(Key{tmax, itr.Tags()}, values)
+	}
+}
+
+// ReduceStddev computes the stddev of values.
+func ReduceStddev(key Key, values []interface{}, e *Emitter) {
+	var data []float64
+	// Collect all the data points
+	for _, value := range values {
+		data = append(data, value.([]float64)...)
+	}
+	// If no data, leave
+	if len(data) == 0 {
+		return
+	}
+	// If we only have one data point, the std dev is undefined
+	if len(data) == 1 {
+		e.Emit(key, "undefined")
+		return
+	}
+	// Get the sum
+	var sum float64
+	for _, v := range data {
+		sum += v
+	}
+	// Get the mean
+	mean := sum / float64(len(data))
+	// Get the variance
+	var variance float64
+	for _, v := range data {
+		dif := v - mean
+		sq := math.Pow(dif, 2)
+		variance += sq
+	}
+	variance = variance / float64(len(data)-1)
```

Comment:
Actually, I see you are right. n for "population standard deviation" and n-1 for "sample standard deviation".